### PR TITLE
feat: accordion component cleanup

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses #2527 #2246

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Our accordion component was built within our "Unit Tables" component which is heavily tied to our data model, and built so we couldn't use the accordion anywhere else.

I pulled the accordion into its own component, allowed for SF styling, and made it more accessible.

In a future ticket we should remove the Unit Tables component from ui-components entirely.

## How Can This Be Tested/Reviewed?

Look at the new stories and the accordions on a listing details page.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: ui-components/src/page_components/listing/ContentAccordion.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import React, { useState, useRef } from "react"
+import { Icon, IconFillColors } from "../../icons/Icon"
+import "./ContentAccordion.scss"
+
+interface UnitTablesProps {
```
Comment (YazeedLoonat) [inline ui-components/src/page_components/listing/ContentAccordion.tsx:5]: nitpicky: should this be ContentAccordionProps?

**Review by YazeedLoonat — CHANGES_REQUESTED** [review]
Looks good @emilyjablonski along with the requested changes, there is also a warning around the Accordion needing a key prop

**Comment by emilyjablonski** [comment]
@YazeedLoonat Ready for your eyes again

**Comment by emilyjablonski** [comment]
@YazeedLoonat another poke about this one :)

**Review by YazeedLoonat — APPROVED** [review]
LGTM

**Pull request merged**